### DISCUSSION TITLE
No matching key exchange method found "in gdb via ssh"

### DISCUSSION THREAD

**Discussion by ethergrouik** [NationalSecurityAgency/ghidra#3444] [Debugger]
Hello everyone,

Linux XXX 5.4.0-86-generic #97-Ubuntu
ghidra_10.0.3_PUBLIC

I am not able to use the debugger "in gdb via ssh":
"Could not connect: ghidra.dbg.error.DebuggerModelTerminatingException: Error while starting GDB"

While I look at /var/log/auth.log of the target (Kali 2021.3), I can read this:
"Sep 22 15:07:51 VMkali sshd[1913]: Unable to negotiate with 192.168.56.10 port 60126: no matching key exchange method found. Their offer: diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1 [preauth]"

Does someone know how to remove these key exchange methods?

Thanks for you help

**Comment by d-millar** [comment]
@ethergrouik  I think the discussion in #2974 is relevant.  Because we're using a rather outdated ssh implementation, you'll need to add the following to your target's sshd config:

KexAlgorithms +diffie-hellman-group1-sha1
Ciphers +aes128-cbc

This admittedly is a workaround - a real fix is on our to-do list.
(Thanks @jrmuizel for identifying this problem and the solution)

**Reply by ethergrouik** [reply]
Thanks for your answer @d-millar and thanks also to @jrmuizel.
